humanlayer/core/approval.py

```diff
@@ -399,6 +401,34 @@ def fetch_approval(
                 continue
             return FunctionCall.Completed(call=call)
 
+    async def create_function_call_ws(
+        self,
+        spec: FunctionCallSpec,
+        call_id: str | None = None,
+    ) -> FunctionCall:
+        call_id = call_id or self.genid("call")
+        call = FunctionCall(
+            run_id=self.run_id,  # type: ignore
+            call_id=call_id,
+            spec=spec,
+        )
+
+        async with websockets.connect("ws://localhost:8000/ws") as websocket:
```

Comment:
What happened when it reached the server, and why is it connecting to localhost?